File: src/main/java/com/codeborne/json/JsonParser.java
package com.codeborne.json;

import org.intellij.lang.annotations.Language;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.Reader;
import java.io.StringReader;

/**
 * <a href="https://www.json.org/json-en.html">JSON specification</a>
 */
public class JsonParser {
  public Object parse(@Language("JSON") String input) throws IOException, JsonParseException {
    return parse(new StringReader(input));
  }

  public Object parse(Reader input) throws IOException, JsonParseException {
    BufferedReader bufferedReader = new BufferedReader(input);
    String line;
    StringBuilder result = new StringBuilder();
    while ((line = bufferedReader.readLine()) != null) {
      result.append(line);
    }
    String string = result.toString();
    System.out.println(string);

    switch (string) {
      case "null":
        return null;
      default:
        return Boolean.parseBoolean(string);
    }

  }
}

File: src/main/java/com/codeborne/json/JsonParser.kt
package com.codeborne.json

import org.intellij.lang.annotations.Language
import java.io.Reader
import java.io.StringReader

/**
 * [JSON specification](https://www.json.org/json-en.html)
 */
class JsonParser {
    fun parse(@Language("JSON") input: String?): Any? {
        return parse(StringReader(input))
    }

    @Throws(IOException::class, JsonParseException::class)
    fun parse(@Language("JSON") input: Int?): Any? {
        return input
    }

    @Throws(IOException::class, JsonParseException::class)
    fun parse(input: Reader?): Any? {
        return input?.let {
            val text = it.readText()
            if (text == "null") return null
            text.toBoolean()
        }
    }
}
